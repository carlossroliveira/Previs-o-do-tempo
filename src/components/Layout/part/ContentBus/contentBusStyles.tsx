import styled from 'styled-components';

export const ContainerSC = styled.section`
  grid-area: contentBus;
  color: white;
`;
