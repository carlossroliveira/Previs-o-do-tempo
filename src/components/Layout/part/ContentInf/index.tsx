// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { SwitchComponent } from '../../../SwitchComponent';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, TitleSC } from './contentInfStyles';

export const ContentInf = (): JSX.Element => {
  return (
    <ContainerSC>
      <TitleSC>Tema</TitleSC>
      <SwitchComponent />
    </ContainerSC>
  );
};
