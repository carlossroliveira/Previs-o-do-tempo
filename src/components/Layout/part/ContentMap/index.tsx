// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './ContentMapStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const ContentMap = (): JSX.Element => {
  return <ContainerSC>MAPA</ContainerSC>;
};
