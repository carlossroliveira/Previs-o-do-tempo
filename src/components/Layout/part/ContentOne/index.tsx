// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './contentOneStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const ContentOne = (): JSX.Element => {
  return <ContainerSC>CONTENTONE</ContainerSC>;
};
