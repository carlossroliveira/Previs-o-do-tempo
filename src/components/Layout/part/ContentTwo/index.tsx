// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './contentTwoStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const ContentTwo = (): JSX.Element => {
  return <ContainerSC>CONTENTTWO</ContainerSC>;
};
