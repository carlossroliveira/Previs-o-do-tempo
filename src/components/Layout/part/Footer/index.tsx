// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './footerStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const Footer = (): JSX.Element => {
  return (
    <ContainerSC>
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações{' '}
    </ContainerSC>
  );
};
