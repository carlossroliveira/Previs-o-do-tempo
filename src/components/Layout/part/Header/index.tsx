// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './headerStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const Header = (): JSX.Element => {
  return <ContainerSC>HEADER</ContainerSC>;
};
