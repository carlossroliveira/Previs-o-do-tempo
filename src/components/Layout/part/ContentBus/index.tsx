// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Button } from '../../../Button';
import { Input } from '../../../Input';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './contentBusStyles';

export const ContentBus = (): JSX.Element => {
  return (
    <ContainerSC>
      <Input />
      <Button />
    </ContainerSC>
  );
};
