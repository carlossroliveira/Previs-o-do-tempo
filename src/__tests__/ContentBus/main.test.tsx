// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';

import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import theme from '../../styles/themes/dark';
import { ContentBus } from '../../components/Layout/part/ContentBus';
// -------------------------------------------------
// Utils
// -------------------------------------------------
import WithThemeComponent from '../utils/withThemeComponent';

const ContentBusWithTheme = WithThemeComponent(ContentBus);

describe('ContentBus Testing', () => {
  it('Should render the ContentBus component - with input and button', () => {
    render(<ContentBusWithTheme />);
    screen.getByRole('textbox');
    screen.getByText(/buscar/i);
  });

  it('should expect the paragraph to equal null from the beginning', () => {
    render(<ContentBusWithTheme />);
    expect(screen.queryByText(/Digite apenas texto/i)).toBeNull();
  });

  it('should render the ContentBus component with the right color selected', () => {
    render(<ContentBusWithTheme />);
    expect(screen.getByRole('textbox')).toHaveStyle({
      color: theme.color.secondary,
    });
  });

  it('should match snapshot', () => {
    const component = render(<ContentBusWithTheme />);
    expect(component).toMatchSnapshot();
  });
});
