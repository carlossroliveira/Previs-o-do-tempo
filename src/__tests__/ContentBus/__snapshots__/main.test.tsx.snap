// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBus Testing should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c3 {
  display: grid;
  place-items: center;
}

.c4 {
  font-size: 2rem;
  cursor: pointer;
  position: relative;
  padding: 0.25em 1em;
  border-radius: 0.25em;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #EEEEEE;
  border: #EEEEEE 0.125em solid;
  text-shadow: 0 0 0.125em hsl(0 0% 100% / 0.3), 0 0 0.45em #EEEEEE;
  box-shadow: inset 0 0 0.5em 0 white,0 0 0.5em 0 white;
}

.c4::before {
  content: '';
  position: absolute;
  pointer-events: none;
  top: 120%;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.7;
  -webkit-filter: blur(1em);
  filter: blur(1em);
  background-color: #EEEEEE;
  -webkit-transform: perspective(1em) rotateX(40deg) scale(1,0.35);
  -ms-transform: perspective(1em) rotateX(40deg) scale(1,0.35);
  transform: perspective(1em) rotateX(40deg) scale(1,0.35);
}

.c4:hover,
.c4:focus {
  color: #393E46;
  background-color: #EEEEEE;
  text-shadow: none;
}

.c4:hover::before,
.c4:focus::before {
  opacity: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
  border-radius: 20px;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
  -ms-transform: scale(0.5);
  -webkit-transform: scale(0.5);
  background-color: #EEEEEE;
  box-shadow: 0 10px 40px #EEEEEE, 0 0 0 20px #393E46, 0 0 0 20px #EEEEEE;
  box-shadow: 0 10px 40px #eeeeee,0 0 0 20px #393e46,0 0 0 20px #eeeeee;
}

.c1:hover {
  box-shadow: 0 10px 40px #393E46, 0 0 0 20px #EEEEEE, 0 0 0 20px #EEEEEE;
}

.c2 {
  text-indent: 4rem;
  padding: 0;
  margin: 0;
  border: 0;
  outline: none;
  background-color: transparent;
  color: #393E46;
  width: 100%;
  height: 96px;
  font-size: 60px;
  line-height: 1;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  grid-area: contentBus;
  margin-bottom: 5rem;
  display: grid;
  color: #EEEEEE;
  font-family: Roboto,sans-serif;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
  -webkit-animation: eQudGy 0.5s linear forwards;
  animation: eQudGy 0.5s linear forwards;
}

<body>
    <div>
      <section
        class="c0"
      >
        <div
          class="c1"
        >
          <input
            class="c2"
            placeholder="Digite seu estado"
            type="text"
            value=""
          />
        </div>
        <div
          class="c3"
        >
          <a
            class="c4"
          >
            Buscar
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": .c3 {
  display: grid;
  place-items: center;
}

.c4 {
  font-size: 2rem;
  cursor: pointer;
  position: relative;
  padding: 0.25em 1em;
  border-radius: 0.25em;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #EEEEEE;
  border: #EEEEEE 0.125em solid;
  text-shadow: 0 0 0.125em hsl(0 0% 100% / 0.3), 0 0 0.45em #EEEEEE;
  box-shadow: inset 0 0 0.5em 0 white,0 0 0.5em 0 white;
}

.c4::before {
  content: '';
  position: absolute;
  pointer-events: none;
  top: 120%;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.7;
  -webkit-filter: blur(1em);
  filter: blur(1em);
  background-color: #EEEEEE;
  -webkit-transform: perspective(1em) rotateX(40deg) scale(1,0.35);
  -ms-transform: perspective(1em) rotateX(40deg) scale(1,0.35);
  transform: perspective(1em) rotateX(40deg) scale(1,0.35);
}

.c4:hover,
.c4:focus {
  color: #393E46;
  background-color: #EEEEEE;
  text-shadow: none;
}

.c4:hover::before,
.c4:focus::before {
  opacity: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
  border-radius: 20px;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
  -ms-transform: scale(0.5);
  -webkit-transform: scale(0.5);
  background-color: #EEEEEE;
  box-shadow: 0 10px 40px #EEEEEE, 0 0 0 20px #393E46, 0 0 0 20px #EEEEEE;
  box-shadow: 0 10px 40px #eeeeee,0 0 0 20px #393e46,0 0 0 20px #eeeeee;
}

.c1:hover {
  box-shadow: 0 10px 40px #393E46, 0 0 0 20px #EEEEEE, 0 0 0 20px #EEEEEE;
}

.c2 {
  text-indent: 4rem;
  padding: 0;
  margin: 0;
  border: 0;
  outline: none;
  background-color: transparent;
  color: #393E46;
  width: 100%;
  height: 96px;
  font-size: 60px;
  line-height: 1;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 {
  grid-area: contentBus;
  margin-bottom: 5rem;
  display: grid;
  color: #EEEEEE;
  font-family: Roboto,sans-serif;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
  -webkit-animation: eQudGy 0.5s linear forwards;
  animation: eQudGy 0.5s linear forwards;
}

<div>
    <section
      class="c0"
    >
      <div
        class="c1"
      >
        <input
          class="c2"
          placeholder="Digite seu estado"
          type="text"
          value=""
        />
      </div>
      <div
        class="c3"
      >
        <a
          class="c4"
        >
          Buscar
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
